src/main/java/com/cognifide/aemrules/utils/MultiMap.java

```diff
@@ -0,0 +1,96 @@
+/*-
+ * #%L
+ * AEM Rules for SonarQube
+ * %%
+ * Copyright (C) 2015-2019 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.aemrules.utils;
+
+import java.util.*;
+import java.util.function.Consumer;
+
+/**
+ * @link https://gist.github.com/kendfinger/9007632
```

Comment:
The code seems to be copied from a Gist without a license. Code without an explicit license is protected by copyright and is by default All Rights Reserved. We can't use this here.
On a related note, why do we need a custom multimap implementation?

Reply:
seems that you you are not right - https://softwareengineering.stackexchange.com/questions/302555/what-is-the-license-of-a-github-gist?noredirect=1&lq=1#comment735292_302555

yep it is not explicit, but implicitly...

https://help.github.com/en/github/writing-on-github/creating-gists

> Every gist is a Git repository, which means that it can be forked and cloned

https://help.github.com/en/github/creating-cloning-and-archiving-repositories/licensing-a-repository

> However, without a license, the default copyright laws apply, meaning that you retain all rights to your source code and no one may reproduce, distribute, or create derivative works from your work

Reply:
I think you misread the terms. The second line you quoted specifically says we *cannot* use this code :)

> without a license, the default copyright laws apply, meaning that you retain all rights to your source code and no one may reproduce, distribute, or create derivative works from your work

Reply:
yep I missed "no one" ;)  you are right.

so decided to rewrite it to simple `Map<String, List<String>>` instead of `Multimap` 

done.